main.go

```diff
@@ -42,88 +45,408 @@ func main() {
 		os.Exit(1)
 	}
 
 	// Output the generated review
-	fmt.Println("Generated Review:")
-	fmt.Println(review)
+	log.Println("------- Generated Review:")
+	log.Println(review)
+	log.Println(`------- File comments:`)
+
+	for _, comment := range reviewComments {
+		log.Printf("File: %s, Line: %d\nComment: %s\n", *comment.Path, *comment.Position, *comment.Body)
+	}
+
+	log.Println(`-------`)
 
-	// Post the review if a flag is set
-	if *approve || *requestChanges {
-		err = postReview(client, ctx, *owner, *repo, *prNumber, review, *approve)
+	if *dryRun {
+		// If dry run, don't post the review, just output the results
+		log.Println("Dry run: Review not posted to GitHub.")
+		return
+	}
+
+	// Check if the reviewer is the PR author
+	isSelfReview := user.GetLogin() == pr.User.GetLogin()
+
+	if isSelfReview {
+		// Post the review as a comment instead
+		commentBody := review
+		if action == "approve" {
+			commentBody += "\n\n**Note:** This is a self-approved PR."
+		} else if action == "request_changes" {
+			commentBody += "\n\n**Note:** This is a self-requested change."
+		}
+
+		// Use the PullRequests.CreateReview method to post review comments directly on lines
+		reviewEvent := &github.PullRequestReviewRequest{
+			Body:     github.String(commentBody),
+			Event:    github.String("COMMENT"), // "COMMENT" will not change the state of the PR
+			Comments: reviewComments,           // Use the existing review comments
+		}
+
+		_, _, err := client.PullRequests.CreateReview(ctx, *owner, *repo, *prNumber, reviewEvent)
 		if err != nil {
-			fmt.Printf("Error posting review: %v\n", err)
-			os.Exit(1)
+			log.Fatalf("\n\n GH Review self-review comments: %v\n", err)
+		}
+
+		fmt.Println("Self-review posted as a comment.")
+	} else {
+		// Determine the action based on the assistant's recommendation and PR checks
+		var state string
+		if action == "approve" && checksPassed {
+			state = "APPROVE"
+		} else if action == "request_changes" || !checksPassed {
+			state = "REQUEST_CHANGES"
+		} else {
+			fmt.Println("Assistant recommended approval, but tests are failing. Requesting changes instead.")
+			state = "REQUEST_CHANGES"
+		}
+
+		// Post the review if not a dry run
+		err = postReviewWithComments(client, ctx, *owner, *repo, *prNumber, review, reviewComments, state)
+		if err != nil {
+			log.Fatal("Error posting review: %v\n", err)
 		}
 		fmt.Println("Review posted successfully!")
 	}
 }
 
-// generateReviewWithAssistant sends PR details to the OpenAI assistant and gets a review summary
-func generateReviewWithAssistant(pr *github.PullRequest) (string, error) {
+// getPendingReview checks if there's a pending review for the PR
+func getPendingReview(client *github.Client, ctx context.Context, owner, repo string, prNumber int) (*github.PullRequestReview, error) {
+	reviews, _, err := client.PullRequests.ListReviews(ctx, owner, repo, prNumber, &github.ListOptions{})
+	if err != nil {
+		return nil, err
```

Comment:
The function createChatCompletion should handle rate limiting and retries for the OpenAI API.